spring-boot-project/spring-boot-actuator-autoconfigure/src/test/java/org/springframework/boot/actuate/autoconfigure/metrics/JvmMetricsAutoConfigurationTests.java

```diff
@@ -43,41 +49,53 @@
 
 	@Test
 	void autoConfiguresJvmMetrics() {
-		this.contextRunner.run(
-				(context) -> assertThat(context).hasSingleBean(JvmGcMetrics.class).hasSingleBean(JvmMemoryMetrics.class)
-						.hasSingleBean(JvmThreadMetrics.class).hasSingleBean(ClassLoaderMetrics.class));
+		this.contextRunner.run((context) -> validateBaseJvmMetricsBeansArePresent());
 	}
 
 	@Test
 	void allowsCustomJvmGcMetricsToBeUsed() {
-		this.contextRunner.withUserConfiguration(CustomJvmGcMetricsConfiguration.class)
-				.run((context) -> assertThat(context).hasSingleBean(JvmGcMetrics.class).hasBean("customJvmGcMetrics")
-						.hasSingleBean(JvmMemoryMetrics.class).hasSingleBean(JvmThreadMetrics.class)
-						.hasSingleBean(ClassLoaderMetrics.class));
+		this.contextRunner.withUserConfiguration(CustomJvmGcMetricsConfiguration.class).run((context) -> {
+			validateBaseJvmMetricsBeansArePresent();
```

Comment:
Shouldn't the code be called like below for it to actually run?
```suggestion
			validateBaseJvmMetricsBeansArePresent().accept(context);
```